### DISCUSSION TITLE
auth.js and qwik - Problem with adding the user role in the JWT

### DISCUSSION THREAD

**Discussion by gimonaa** [nextauthjs/next-auth#11672] [Help]
### Environment

```
  System:
    OS: Linux 5.10 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (4) x64 Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz
    Memory: 968.16 MB / 5.79 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 20.10.0 - /usr/local/lib/nodejs/node-v20.10.0-linux-x64/bin/node
    Yarn: 1.22.15 - /usr/bin/yarn
    npm: 10.2.3 - /usr/local/lib/nodejs/node-v20.10.0-linux-x64/bin/npm
    pnpm: 9.0.5 - ~/.local/share/pnpm/pnpm
  npmPackages:
    @auth/core: ^0.34.1 => 0.34.2 
    @auth/qwik: 0.2.2 => 0.2.2 
```

### Reproduction URL

https://github.com/gimonaa/auth.qwik

### Describe the issue

I would like to extend the user by also adding the role in the session. If I install only Qwik and add `@auth/qwik`, I can do this by extending `JWT`, `User`, and `Session` (in the `types` folder). However, if I add Prisma, the `@auth/core` folder is duplicated within `@auth/qwik`, and the extension is no longer recognized, resulting in an error where `role` is not defined.

Property `role` does not exist on type `User | AdapterUser`.

### How to reproduce

if I delete the `@auth/qwik/node_modules` dir and its contents the problem disappears, then with an `npm update` the folder i recreated and the problem reappears

### Expected behavior

the role error should not appear in the JWT callback

**Comment by balazsorban44** [comment]
The issue seems to stem from these:

https://github.com/gimonaa/auth.qwik/blob/82c13d8a06b5f55164f17e26bdb91be1d43743db/package.json#L56

https://github.com/gimonaa/auth.qwik/blob/main/src/types/next-auth.d.ts


First, you should not need to have `@auth/core` installed in your project, only `@auth/qwik`.

Also, the type augmentation seems a bit messy. `next-auth` is not part of a Qwik project, so the import seems odd.

You should not augment `node_modules`, or `@auth/core` types just `@auth/qwik`

**Reply by gimonaa** [reply]
I tried to but @auth/qwik has no exported types, I can't extend User or AdapterUser.
"Property role does not exist on type User | AdapterUser"
The only exports I found are in @auth/qwik/node_modules/@auth/ and I didn't find a way to extend them.

https://github.com/gimonaa/auth.qwik/blob/main/src/types/next-auth.d.ts
this was a workaround I found online that worked with the previous version of qwik auth (@builder.io/qwik-auth)

Now I updated my repo removing next-auth but I don't know how to extend User or AdapterUser to include the role in the session and in JWT.

https://github.com/gimonaa/auth.qwik

Please help

Thanks

**Comment by gimonaa** [comment]
I tried to but @auth/qwik has no exported types, I can't extend User or AdapterUser.
"Property role does not exist on type User | AdapterUser"
The only exports I found are in @auth/qwik/node_modules/@auth/ and I didn't find a way to extend them.

https://github.com/gimonaa/auth.qwik/blob/main/src/types/next-auth.d.ts
this was a workaround I found online that worked with the previous version of qwik auth (@builder.io/qwik-auth)

Now I updated my repo removing next-auth but I don't know how to extend User or AdapterUser to include the role in the session and in JWT.

https://github.com/gimonaa/auth.qwik

Please help

Thanks
